maze_transformer/evaluation/plot_maze.py

```diff
@@ -6,20 +6,113 @@
 
 from maze_transformer.generation.latticemaze import LatticeMaze
 
+CoordArray = NDArray["coords", np.int8]
+
+# Set size of unit.
+# Each "unit" consists of a node and the right and lower adjacent wall/connection. Its area is ul * ul.
+# Wall thickness is fixed to 1, so this parameter sets the ratio of node length to wall thickness
+ul = 14
+
+
+# Maze conversion functions (No plt involved)
+
+
+def as_img(maze: LatticeMaze) -> NDArray["x y", bool]:
+    """
+     Build an image to visualise the maze.
+     Each "unit" consists of a node and the right and lower adjacent wall/connection. Its area is ul * ul.
+     - Nodes are displayed as white squares of area: (ul-1) * (ul-1)
+     - Walls are displayed as black recktangles of area: 1 * (ul-1)
+     - Connections are displayed as light grey or white rectangles of area: 1 * (ul-1); color is depending on show_connections argument
+
+     Axes definition:
+    (0,0)     y
+      -|----------->
+     x |
+       |
+       |
+       v
+
+     Returns a matrix of side length (ul) * n + 1 where n is the number of nodes.
+    """
+
+    # Set color of connections (using plt.imshow(cmap='grey))
+    show_connections = True
+
+    white_color = 100
+    if show_connections:
+        connection_color = white_color * 0.93
+    else:
+        connection_color = white_color
+
+    # Set up the background (walls everywhere)
+    img: NDArray["x y", int] = np.zeros(
+        (
+            maze.grid_shape[0] * ul + 1,
+            maze.grid_shape[1] * ul + 1,
+        ),
+        dtype=int,
+    )
+
+    # Draw nodes and connections by iterating through lattice
+    for x in range(maze.grid_shape[0]):
+        for y in range(maze.grid_shape[1]):
+            # Draw node
+            img[x * ul + 1 : (x + 1) * ul, y * ul + 1 : (y + 1) * ul] = white_color
+
+            # Down connection
+            if maze.connection_list[0, x, y]:
+                img[(x + 1) * ul, y * ul + 1 : (y + 1) * ul] = connection_color
+
+            # Right connection
+            if maze.connection_list[1, x, y]:
+                img[x * ul + 1 : (x + 1) * ul, (y + 1) * ul] = connection_color
+
+    return img
+
+
+def points_transform_to_img(points: CoordArray) -> CoordArray:
+    """transform points to img coordinates"""
+
+    #####
+    # Swapping x, y axes here may be deprecated with resolution of issue #73
+    points = np.array([(x, y) for (y, x) in points])
```

Comment:
Hmm..yeah, we shouldn't need to do this any more. There's something fishy about this

Reply:
You need to do this any time you're passing a list of `Coords` into `plt.plot` or similar functions, because the former use (row, col) array coordinates and the latter use (x, y) Cartesian coordinates, which are opposite.

In this case we definitely shouldn't be doing it - this function returns a `CoordArray` and those should always be (row, col)

Reply:
@afspies thats right, renamed to row / col as suggested